### DISCUSSION TITLE
Exposing app as an audio device

### DISCUSSION THREAD

**Discussion by johnfoxii** [SamDel/ChromeCast-Desktop-Audio-Streamer#106] [Ideas]
Sorry if this is redundant.
I was thinking to allow the app to be selected as an audio output device in windows.
Maybe this would be in the Windows UWP.

**Comment by SamDel** [comment]
Nice idea! It will create new possibilities. I'm thinking on configuring (in the 'App volume and device preferences' widget) which application streams to DAS. What were you thinking on?

I don't know what's the best way to do it. You probably have to write a driver. Do you know how to do it?

**Reply by johnfoxii** [reply]
I  could not write it but I could take on the research and get the right parts collected. I have VS2017 and a google dev account. Lets see if and what I can put together.
Question, do you think there is something to be done with the buffering delay. If you need to have someone run the code feeing it different settings to find the right combo. I would be happy to try.
Thanks

**Reply by SamDel** [reply]
> I could not write it but I could take on the research and get the right parts collected. I have VS2017 and a google dev account. Lets see if and what I can put together.

Nice! Let me know.

> Question, do you think there is something to be done with the buffering delay. If you need to have someone run the code feeing it different settings to find the right combo. I would be happy to try.

See [this issue](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/issues/27) for all the things we tried with latency (if that's what you mean). For me it's < 2.5 seconds with wav 24 bit as stream format. How is the delay for you?

**Reply by pstrg** [reply]
Working - it's possible to send sound from a local application (foobar2000) to a Chromecast Audio device.
However (besides the delay)
1) the sound is quite muffled, undistinct (no exaggeration);
2) at times, it's not possible to select the device: it's necessary to restart.

**Reply by SamDel** [reply]
> 1. the sound is quite muffled, undistinct (no exaggeration)

For me the audio quality is a lot better when I plug in headphones (and stream with the headphones as the recording source). The application doesn't change the audio before streaming.

 >2. at times, it's not possible to select the device: it's necessary to restart.

Doesn't the device show up, or doesn't it start streaming?

**Comment by johnfoxii** [comment]
Is it Sam?
No matter what the setting is set in the app I have a delay of  about 17
seconds starting or stopping.
using the casting built into youtube music or spotify I sometimes notice a
1 or 2 second delay.
Tracing the cast in YouTubeMusic I see it using Windows Media Foundation
I cloned your code. a quick look i didn't see these libraries used but i
only had a minute
more soon

**Comment by SamDel** [comment]
< 2.5 seconds is the delay between the audio on the desktop and the audio on the speakers.
17 seconds is way too much. Can you post the log, then I can have a look.

**Comment by waskel** [comment]
Great app, but I'm having a delay of 10-12 seconds. Seems long?

... never mind. Changed to WAV, it dropped down to about 1-1.5 seconds.

**Comment by pstrg** [comment]
Hi!

I understand that redirecting shouldn't affect the sound, but it does -
don't imagine why.
It's not related to the use of either speakers or phones.

Devices always show up; at times they just don't turn green even after a
while.

**Reply by SamDel** [reply]
I have no clue why the sound quality is so poor.

When you post the log I can have a look why the device doesn't turn green!
